### DISCUSSION TITLE
Naming features

### DISCUSSION THREAD

**Discussion by ogregoire** [bagelbits/5e-database#382] [General]
The PR #380 has me notice that we still have some naming inconsistencies.

I believe that we settled the debate about containing "Choose: " and that we should remove it because the real SRD name doesn't contain that.

However if we keep this, I wonder why we still keep "Expertise 1" for bards, for instance, instead of "Expertise". To me, we should be as close as possible to the SRD, and nowhere does the SRD mention the "Expertise 1" or "Metamagic, First and Second": only "Expertise" (SRD, pp. 13, 39) and "Metamagic" (SRD, p. 44) are mentioned, and if you look at the class tables, it's the same: only the short name is ever written.

So I believe that we should check the SRD and keep names as close as possible to it.

But the question will rise about the indices: they also still contain the "-choose-" part and this part should be replaced by a single "-". Given that index are kind of elements of a URL, I understand they should be unique, and I believe that we should keep them as unique as possible. And here, I'm inclined to use the 1 and 2 because it just makes sense because it needs to be unique.

And for the index, I suggest actually the following format:

    [ <class name> - ] feature [ - <feature unique number> ]

Where `[ <class name> - ]` is optional and must be added only when a feature name exist in two different classes (even if they're different just like bard's expertise doesn't include thieves tools while rogue's expertise does include it), and `[ - <feature unique number> ]` is an increasing number starting from 1 when a feature for a same class is given at various levels, no matter the quantity of options given each level.

So my suggestion is to have the following example:

| Class | Level  | Current name                       | Current index                            | Proposed name | Proposed index | 
| ----- | ------ | ----------------------------- | -------------------------------  | -------------- | -------------- | 
| Bard  | 3        | Expertise 1                            | `bard-choose-expertise-1`        | Expertise        | `bard-expertise-1` |
| Bard | 10       | Expertise 2                            | `bard-choose-expertise-2`        | Expertise        | `bard-expertise-2` |
| Fighter | 1     | Fighting Style                        | `fighter-choose-fighting-style`  | Fighting Style | `fighter-fighting-style` |
| Paladin | 1     | Fighting Style                        | `paladin-choose-fighting-style` | Fighting Style | `paladin-fighting-style` |
| Ranger | 1     | Fighting Style                        | `ranger-choose-fighting-style`  | Fighting Style | `ranger-fighting-style` |
| Rogue | 1      | Expertise 1                            | `rogue-choose-expertise-1`       | Expertise        | `rogue-expertise-1` |
| Rogue | 6      | Expertise 2                            | `rogue-choose-expertise-2`       | Expertise        | `rogue-expertise-2` |
| Sorcerer | 1   | Metamagic, First and Second | `choose-metamagic`                 | Metamagic     | `metamagic-1` |
| Sorcerer | 10 | Metamagic, Third                   | `choose-additional-metamagic-1` | Metamagic      | `metamagic-2` |
| Sorcerer | 17 | Metamagic, Fourth                | `choose-additional-metamagic-2` | Metamagic     | `metamagic-3` |

There are others, but those examples cover all cases, I guess.

What are your thoughts?

**Comment by bagelbits** [comment]
I really like that as it mirrors the SRD but is functionally the same.

**Comment by Wyzards** [comment]
What are your thoughts on how to handle descriptions?

For example, the current description of `choose-metamagic` is as follows:
`"At 3rd level, you gain the ability to twist your spells to suit your needs. You gain two of the following Metamagic options of your choice. You gain another one at 10th and 17th level.",
"You can use only one Metamagic option on a spell when you cast it, unless otherwise noted."`

Do you think the description should be the same for the `choose-additional-mektamagic-1`, and so on features?

Regardless, I agree with the changes listed here and would be happy to implement them.

**Reply by ogregoire** [reply]
Yes, I believe so. We aren't supposed to edit the SRD: only to transcribe it.

**Comment by Wyzards** [comment]
Also your current listed indexes for metamagic are incorrect.

They are as follows:
- `choose-metamagic`
- `choose-additional-metamagic-1`
- `choose-additional-metamagic-2`...

Should these be changes to `metamagic-1`, `metamagic-2`, etc.?

**Reply by ogregoire** [reply]
I fixed it, sorry. My idea is indeed that we change it to `metamagic-1` and friends.
